axelrod/graph.py

```diff
@@ -0,0 +1,126 @@
+"""
+Weighted undirected sparse graphs.
+
+Original source:
+https://github.com/marcharper/stationary/blob/master/stationary/utils/graph.py
```

Comment:
Not against this. A couple of initial thoughts:

1. As you have built this for your stationary library you no doubt have a reason but why we wouldn't just use networkx (which would have the quick lookups we want here)? (1 negative side is adding another dependency to the library) 
2. Can we add some tests please?
3. The spatial tournament does not necessarily need a graph object (it just loops through the edges once via the match generator) but if we're going to have an `axelrod.graph` class we should also use that there (for consistency). I see two approaches:
     1. We allow the spatial tournament to take both a list of edges of one of these graph objects (ensures backwards compatibility);
     2. We let the Moran process take a list of edges and obfuscate the building of this underlying graph object there. I think this would be my preferred approach.